src/main/java/com/coveo/spillway/limit/LimitBuilder.java

```diff
@@ -92,30 +99,42 @@ private LimitBuilder() {}
    * Adds a call back that will be called when the specified limit (using {@link #to(int)}) is reached.
    *
    * @param limitTriggerCallback The callback {@link LimitTriggerCallback}
-   * @return The current LimitBuilder
+   * @return The current {@link LimitBuilder}
    */
   public LimitBuilder<T> withExceededCallback(LimitTriggerCallback limitTriggerCallback) {
     triggers.add(new ValueThresholdTrigger(limitCapacity, limitTriggerCallback));
     return this;
   }
 
   /**
+   * Adds a {@link LimitOverride} to the current {@link Limit}.
+   *
+   * @param limitOverride The {@link LimitOverride}
+   * @return The current {@link LimitBuilder}
+   */
+  public LimitBuilder<T> withLimitOverride(LimitOverride limitOverride) {
```

Comment:
Wouldn't replacing it with a Set ruin the order? :(

You could use a sorted set but it seems even less intuitive.
Unless you mean that you'll use a set that ensures that we have only 1 override per property value?